src/client.ts

```diff
@@ -1410,34 +1402,33 @@ export class MongoClient<U extends BaseMongoObject, L extends BaseMongoObject> {
 					);
 
 					await newValuesStream.forEachPage(async (newValuesEntities: U[]) => {
-						const modifications: { snapshot: U; newEntity: U }[] = [];
+						const snapshotToHistories: U[] = [];
 
 						for (const newEntity of newValuesEntities) {
 							const snapshot: U = oldValuesSaved[newEntity._id];
 
 							if (snapshot) {
-								if (!this.historicManager.areValuesEquals(snapshot, newEntity)) {
-									modifications.push({
-										snapshot,
-										newEntity,
-									});
+								if (!this.areEntitiesEqualToUpdateOnlyOnChange(snapshot, newEntity)) {
+									snapshotToHistories.push(snapshot);
 								}
 							}
 						}
 						if (this.conf.keepHistoric) {
-							await this.historicManager.insertMany(
-								modifications.map((modification) => modification.snapshot),
-								now,
-								userId,
-							);
+							await this.historicManager.insertMany(snapshotToHistories, now, userId);
 						}
 
 						if (this.conf.updateOnlyOnChange) {
-							await this.updateLastModificationDateBulk(modifications, now, userId);
+							await this.updateLastModificationDateBulk(
+								snapshotToHistories.map((value) => value._id),
+								now,
+								userId,
+							);
 						}
 					});
 				}
-				return this.findWithType(newValuesQuery, this.type, 0, newEntities?.length);
+				if (returnNewEntities) {
+					return this.findWithType(newValuesQuery, this.type, 0, newEntities?.length);
```

Comment:
```suggestion
					return this.findWithType(newValuesQuery, this.type, 0, newEntities.length);
```
Not specialy related to this PR, but in fact, newEntities is always defined due to line 1291.

Reply:
Done